### DISCUSSION TITLE
Matrix-based conditional rules

### DISCUSSION THREAD

**Discussion by rizalgunawan** [vueform/vueform#365] [Questions]
How can I implement conditional rules for a matrix component?
Specifically, I want to dynamically change the input type in each row based on the selected value.

<img width="972" alt="image" src="https://github.com/user-attachments/assets/6096d5d2-e38d-4099-ac87-7b457db1b00a">

**Comment by adamberecz** [comment]
Unfortunately you can't reference cells with conditions, let alone changing the element type.

You might be able to achieve this using a nested List and having two fields for static/dynamic with opposite conditions.
